### PR TITLE
[FEATURE] #97700 - Adopt Symfony Messenger as a message bus and queue

### PR DESCRIPTION
**Pull request by brotkrueml**
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/335
Releases: main

### CODE REVIEW

File: Documentation/ApiOverview/MessageBus/Index.rst [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        :caption: EXT:my_extension/Configuration/Services.yaml
+
+"Everyday" usage - as a system administrator/integrator
+=======================================================
+
+By default, TYPO3 will behave like in versions before TYPO3 v12. This means that
+the message bus will use the synchronous transport and all messages will be
+handled immediately. To benefit from the message bus, it is recommended to
+switch to an asynchronous transport. Using asynchronous transports increases the
+resilience of the system by decoupling external dependencies even further.
```
Comment (DavidBruchmann) [inline Documentation/ApiOverview/MessageBus/Index.rst:76]: why "even further", decoupling wasn't mentioned before

File: Documentation/ApiOverview/MessageBus/Index.rst [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+Currently, the TYPO3 Core provides an asynchronous transport based on the
+`Doctrine DBAL messenger transport`_. This transport is configured to use the
+default TYPO3 database connection. It is pre-configured and can be used by
+changing the settings:
+
+..  code-block:: php
+    :caption: config/settings.php | config.additional.php
+
+    $GLOBALS['TYPO3_CONF_VARS']['SYS']['messenger']['routing']['*'] = 'doctrine';
```
Comment (DavidBruchmann) [inline Documentation/ApiOverview/MessageBus/Index.rst:86]: Is here a hint possible how to limit it to extensions or based on different aspects?
In detail I mean what to enter if no wildcard?

File: Documentation/ApiOverview/MessageBus/Index.rst [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    WantedBy=multi-user.target
+
+
+Advanced usage
+==============
+
+Configure a custom transport (Senders/Receivers)
+------------------------------------------------
+
+Transports are configured in the services configuration. To allow the
```
Comment (DavidBruchmann) [inline Documentation/ApiOverview/MessageBus/Index.rst:172]: Is this related to services configuration inside an extension?
If not (like I have the impression reading the following text), perhaps the contradiction should be outlined.

File: Documentation/ApiOverview/MessageBus/Index.rst (hunk 2 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/MessageBus/Index.rst:86]: It is a draft. Next step is to add the new configurations to the according chapters and link to them.

File: Documentation/ApiOverview/MessageBus/Index.rst (hunk 3 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/MessageBus/Index.rst:172]: It is outlined in the following code snippet. There is a caption which explains where to put that part.
TLDR: in my_extension/Configuration/Services.yaml or config/system/services.yaml.

File: Documentation/ApiOverview/MessageBus/Index.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/MessageBus/Index.rst:84]: A message bus decouples the code by using it. The class dispatches a message and thats it (for example for sending an email). The class itself does not handling the sending of the email, but the messenger component does. So the code it decoupled - also with the synchronous transport.

File: Documentation/ApiOverview/MessageBus/Index.rst [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+..  index:: Message Bus
+..  _message-bus:
+
+===========
+Message bus
+===========
+
+..  versionadded:: 12.2
+
+TYPO3 provides a message bus solution based on `symfony/messenger`_. It has the
```
Comment (linawolf) [inline Documentation/ApiOverview/MessageBus/Index.rst:11]: shall we link this to packagist maybe?

Comment (linawolf) [reply Documentation/ApiOverview/MessageBus/Index.rst:11]: Oh ok, link is below

**Review by linawolf — DISMISSED** [review]

**Comment by linawolf** [comment]
@DavidBruchmann is anything still open from your point of view?

**Review by DavidBruchmann — APPROVED** [review]
@linawolf Everything is fine, I think ;-)

File: Documentation/Configuration/Typo3ConfVars/SYS.rst [hunk 5]
```diff
@@ -984,3 +984,67 @@ passwordPolicies
...
+    :ref:`messenger component <message-bus>`. By default, TYPO3 uses a
+    synchronous transport (:php:`default`) for all messages (:php:`*`):
+
+    ..  code-block:: php
+
+        $GLOBALS['TYPO3_CONF_VARS']['SYS']['messenger']['routing'] = [
+            '*' => 'default',
+        ];
+
+    You can set a different transport for a specific message, for example:
```
Comment (brotkrueml) [inline Documentation/Configuration/Typo3ConfVars/SYS.rst:1020]:
```suggestion
    You can set a different transport for a specific message (additional to the above), for example:
```

File: Documentation/Configuration/Typo3ConfVars/SYS.rst [hunk 6]
```diff
@@ -984,3 +984,67 @@ passwordPolicies
...
+
+    ..  code-block:: php
+
+        $GLOBALS['TYPO3_CONF_VARS']['SYS']['messenger']['routing'] = [
+             \MyVendor\MyExtension\Queue\Message\DemoMessage::class => 'doctrine';
+        ];
```
[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/Configuration/Typo3ConfVars/SYS.rst:1026]: Clarified it that it is additional.

File: Documentation/Configuration/Typo3ConfVars/SYS.rst [hunk 7]
```diff
@@ -1017,7 +1017,7 @@ routing
             '*' => 'default',
         ];
```
Comment (DavidBruchmann) [inline Documentation/Configuration/Typo3ConfVars/SYS.rst:1018]: $GLOBALS['TYPO3_CONF_VARS']['SYS']['messenger']['routing']['*'] = 'default';

File: Documentation/Configuration/Typo3ConfVars/SYS.rst [hunk 8]
```diff
@@ -1017,7 +1017,7 @@ routing
...
 
-    You can set a different transport for a specific message, for example:
+    You can set a different transport for a specific message (additional to the above), for example:
 
     ..  code-block:: php
```
Comment (DavidBruchmann) [inline Documentation/Configuration/Typo3ConfVars/SYS.rst:1023]: $GLOBALS['TYPO3_CONF_VARS']['SYS']['messenger']['routing'][\MyVendor\MyExtension\Queue\Message\DemoMessage::class] = 'doctrine';
        
Sorry, couldn't write it better, it didn't let me

**Review by DavidBruchmann — CHANGES_REQUESTED** [review]
"additional" doesn't work if the array is overwritten by PHP, so the notation got quite important now.

File: Documentation/Configuration/Typo3ConfVars/SYS.rst (hunk 8 above)
Comment (brotkrueml) [reply Documentation/Configuration/Typo3ConfVars/SYS.rst:1023]: Aah, you are right, missed it, will adjust it.

Comment (brotkrueml) [reply Documentation/Configuration/Typo3ConfVars/SYS.rst:1023]: Done

File: Documentation/Configuration/Typo3ConfVars/SYS.rst (hunk 7 above)
Comment (brotkrueml) [reply Documentation/Configuration/Typo3ConfVars/SYS.rst:1018]: Not necessary, as this snippet shows the default configuration.

**Review by DavidBruchmann — APPROVED** [review]
